include/zenoh.h

```diff
@@ -33,12 +33,12 @@ extern "C" {
 #include "zenoh_concrete.h"
 #include "zenoh_opaque.h"
 #include "zenoh_commons.h"
+#include "zenoh_macros.h"
```

Comment:
you should not put zenoh-macros.h inside extern "C" block, since they contain templates definitions for C++, which are not allowed inside extern "C".